src/DETHRACE/common/utility.c

```diff
@@ -144,45 +142,52 @@ void EncodeLine(char* pS) {
         }
     }
     while (len > 0 && (pS[len - 1] == '\r' || pS[len - 1] == '\n')) {
-        --len;
-        pS[len] = 0;
+        len--;
+        pS[len] = '\0';
     }
 
     seed = len % 16;
+
     for (i = 0; i < len; i++) {
         c = pS[i];
+#if defined(DETHRACE_FIX_BUGS)
```

Comment:
add a comment documenting the original bug behavior?

Reply:
The OG game (carm95.exe at least) does not do this.
It keeps the original key and does not decode the comments with the alternative key.
I'll add this as a comment.

Reply:
that would be great, thanks.